src/com/facebook/buck/jvm/groovy/GroovycStep.java

```diff
@@ -91,14 +98,63 @@ public String getDescription(ExecutionContext context) {
   }
 
   private ImmutableList<String> createCommand() {
-    ImmutableList.Builder<String> command = ImmutableList.builder();
-    command.addAll(groovyc.getCommandPrefix(resolver))
+    final ImmutableList.Builder<String> command = ImmutableList.builder();
+
+    command.addAll(groovyc.getCommandPrefix(resolver));
+
+    String classpath = Joiner.on(File.pathSeparator).join(transform(declaredClasspathEntries, toStringFunction()));
+    command
         .add("-cp")
-        .add(Joiner.on(":").join(transform(declaredClasspathEntries, toStringFunction())))
+        .add(classpath.isEmpty() ? "''" : classpath)
         .add("-d")
-        .add(outputDirectory.toString())
-        .addAll(extraArguments.or(ImmutableList.<String>of()))
-        .addAll(transform(sourceFilePaths, toStringFunction()));
+        .add(outputDirectory.toString());
+    addCrossCompilationOptions(command);
+
+    command.addAll(extraArguments.or(ImmutableList.<String>of()))
+           .addAll(transform(sourceFilePaths, toStringFunction()));
     return command.build();
   }
+
+  private void addCrossCompilationOptions(final ImmutableList.Builder<String> command) {
+    if (shouldCrossCompile()) {
+      command.add("-j");
+      javacOptions.appendOptionsTo(new OptionsConsumer() {
+        @Override
+        public void addOptionValue(String option, String value) {
+          // Explicitly disallow the setting of sourcepath in a cross compilation context.
+          // The implementation of `appendOptionsTo` provides a blank default, which
+          // confuses the cross compilations step's javac (it won't find any class files
+          // compiled by groovyc).
+          if (!option.equals("sourcepath")) {
+            command.add("-J" + String.format("%s=%s", option, value));
+          }
```

Comment:
This logic might be a tiny bit clearer if you just did a return in the `if` block and afterwards had the common case.